tensorflow/lite/micro/micro_allocator.cc

```diff
@@ -221,21 +221,24 @@ TfLiteStatus AllocationInfoBuilder::GetOfflinePlannedOffsets(
 
 TfLiteStatus AllocationInfoBuilder::AddScratchBuffers(
     internal::ScratchBufferRequest* scratch_buffer_requests,
-    ScratchBufferHandle* scratch_buffer_handles) {
+    ScratchBufferHandle* scratch_buffer_handles, int subgraph_idx) {
```

Comment:
IIUC, AllocationInfoBuilder is created per subgraph (https://source.corp.google.com/piper///depot/google3/third_party/tflite_micro/tensorflow/lite/micro/micro_allocator.cc;l=900;bpv=1;bpt=1;rcl=419362405).

The tensor_count_ inside AllocationInfoBuilder is already set per subgraph at construction time, so WDYT of passing subgraph_idx as a constructor parameter to AllocationInfoBuilder and avoid passing subgraph_idx all the time?

Reply:
It is - I would slightly prefer to keep it this way to keep AllocationInfoBuilder small. If we need to add subgraph_index into another method I think then it should be part of AllocationInfoBuilder. I don't have a strong preference one way or another, though.